button-merchant/src/main/java/com/usebutton/merchant/ConnectionManagerImpl.java

```diff
@@ -153,4 +158,26 @@ private static JSONObject readResponseBody(HttpURLConnection connection)
         reader.close();
         return new JSONObject(response.toString());
     }
+
+    /**
+     * Refreshes the current session if provided in the network response.
+     * If a null session is provided, the Library data is cleared.
+     */
+    private void refreshSessionIfAvailable(@Nullable JSONObject responseBody) {
+        if (responseBody == null) return;
+
+        try {
+            JSONObject metaJson = responseBody.getJSONObject("object").getJSONObject("meta");
+            if (metaJson.has("session")) {
```

Comment:
Could we do `session_id`? Similar logic exists here: https://github.com/button/mobile-api/blob/f77816b2cc492e363762f535e4b3edb545b5540b/routes/pubapi.js#L191